### PR TITLE
[Card Settings] Break card global settings into Blog, Collection and Product Cards

### LINKED ISSUE #1775: [Card Settings] Break card global settings into Product, Collection and Blog Cards
Currently there is one global card setting under theme settings. With this update, we'll break these card settings into product, collection and blog card settings to provide more flexibility to the merchants.

### PR DESCRIPTION
**Pull request by sofiamatulis**
**PR Summary:** 

Before, all card settings were condensed and affected all cards in the theme. With this update, they still live in the global settings, but they are separated by blog, product and collection

It should be working exactly the same, but the settings are now separated

Figma: https://www.figma.com/file/WzjG1Eld7XJBwm8xBjj8Il/Product-Cards?node-id=235%3A16313
Project Log: https://docs.google.com/document/d/1tASZYuY8RUZQcj_vc1S1q5mCeYtmp1gAZieWlyjXuog/edit#

**Why are these changes introduced?**

Fixes #1775 .

**What approach did you take?**

I added new variables for blog, product and collection cards, which are set in `base.css` and removed the cards settings.

**Other considerations**

**Testing steps/scenarios**

Please test the password page as well

_For products_
- [ ] Featured collection
- [ ] Prod recommendations
- [ ] Collection page
- [ ] Collage
- [ ] Search results

_For blog_
- [ ] Search results
- [ ]  Featured blog
- [ ] Main blog

_For collection_
- [ ] Collections list
- [ ] Cart drawer
- [ ] Main list collection
- [ ] Collage

**Demo links**
_Please include a link to a demo store that includes preconfigured sections and settings to allow reviewers to easily test the features you are working on._

- [Editor](https://os2-demo.myshopify.com/admin/themes/128174817302/editor)

**Checklist**
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Notified the [help.shopify.com](https://help.shopify.com) team about updates to theme settings (current contact: Kimli)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: locales/en.default.schema.json
Comment (sofiamatulis) [inline locales/en.default.schema.json:66]: I think I can remove this 🤔 I dont see it being used anywhere but wanted to ensure before

File: assets/base.css
Comment (sofiamatulis) [reply assets/base.css]: I dont see any changes with removing the `contains-card` , so removing it for now 👍

**Review by duygukalaycioglu — APPROVED** [review]
- [x] Tested collection cards setting's impact on collage
- [x] Tested product cards setting's impact on collage 
Both working well and other settings are looking good!

**Review by katycobb — APPROVED** [review]

File: locales/en.default.schema.json
Comment (andrewetchen) [inline locales/en.default.schema.json:51]: > I think I can remove `header__badge` 🤔 I dont see it being used anywhere but wanted to ensure before

I also just checked and I'm also not seeing `header__badge` used anywhere. We can safely remove all instances from product, collection, and blog cards.

File: locales/en.default.schema.json
Comment (andrewetchen) [reply locales/en.default.schema.json:66]: Yep, we should remove `header__badge`

File: locales/en.default.schema.json
Comment (andrewetchen) [inline locales/en.default.schema.json:83]: Here too.

File: snippets/article-card.liquid
Comment (andrewetchen) [inline snippets/article-card.liquid:26]: I don't have very strong feelings towards doing this but I was wondering what your thoughts were on adding `card-wrapper--article` instead of `article-card-wrapper`?

```suggestion
  <div class="card-wrapper card-wrapper--article underline-links-hover">
```

We'd of course need to update the `base.css` selector from `.article-card-wrapper .card` to `.card-wrapper--article .card`

File: snippets/card-collection.liquid
Comment (andrewetchen) [inline snippets/card-collection.liquid:26]: Same suggestion as above but with `card-wrapper--collection` instead of `collection-card-wrapper`

```suggestion
<div class="card-wrapper card-wrapper--collection animate-arrow">
```

We'd of course need to update the `base.css` selector from `.collection-card-wrapper .card` to `.card-wrapper--collection .card`

File: snippets/card-product.liquid
Comment (andrewetchen) [inline snippets/card-product.liquid:33]: Same suggestion as above but with `card-wrapper--product` instead of `product-card-wrapper`

```suggestion
  <div class="card-wrapper card-wrapper--product underline-links-hover">
```

We'd of course need to update the `base.css` selector from `.product-card-wrapper .card` to `.card-wrapper--product .card`

File: config/settings_data.json
Comment (andrewetchen) [inline config/settings_data.json:66]: I know that we don't have much control over how themes are updated etc. but I wonder if we'll encounter any issues when we update our themes with the above new settings 🤔

**Review by andrewetchen — COMMENTED** [review]
Thanks Sofia! Everything looks good and is working great!

Some feedback and questions from my initial pass:

**Review by duygukalaycioglu — DISMISSED** [review]

File: snippets/article-card.liquid
Comment (andrewetchen) [reply snippets/article-card.liquid:26]: Thanks Ken. Yeah, my thoughts were more or less the same as yours -- no strong feelings and you're probably right in thinking that it'd be too much at this point.

**Comment by melissaperreault** [comment]
We'll need to consider associating a theme setting to apply to [Pages for the search results page](https://screenshot.click/16-41-g8q0p-l09jf.png). My recommendation would be to follow Blog posts for now. I know we need to improve those cards https://github.com/Shopify/dawn/issues/1509 but that can be done later.

**Comment by melissaperreault** [comment]
Hey team! Another point of conversation I want to bring up is the split from associating `Cards` theme settings to Collage section for all block type form relying only on _Product cards_ or _Product cards, Collection cards and Media theme settings._

[Video reference](https://screenshot.click/17-53-1jmam-6b355.mp4)

**Key points**
- Visual changes on merchant stores: Merchants doing manual update will end up with visual change on their store. If this is intended, we'll need to be clear in the release notes.
- Opportunity to introduce the changes in 2 steps. 
   - 1 is to keep relation ship with Products cards first regardless of the block type. Merchants could benefit from all the next theme releases with most of the great updates we are shipping to main.
   - 2 would be to separate the associations relevant to the block type. Gives merchants the ability to chose or not to make the update yet and have already benefited from the pass release wins.
- We may ok with this change because of the flexibility it might bring but merchants will have other key decisions to make in order to make the Collage section feel like it was stilled before. e.g. Settings applied to not only the Collage section and impacting other section where it may be unwanted.

File: snippets/article-card.liquid
Comment (sofiamatulis) [reply snippets/article-card.liquid:26]: That makes sense 🤔 I can keep it this way for now and we can change it in the future if needed.

**Comment by sofiamatulis** [comment]
> Hey team! Another point of conversation I want to bring up is the split from associating `Cards` theme settings to Collage section for all block type form relying only on _Product cards_ or _Product cards, Collection cards and Media theme settings._
> 
> [Video reference](https://screenshot.click/17-53-1jmam-6b355.mp4)
> 
> **Key points**
> 
> * Visual changes on merchant stores: Merchants doing manual update will end up with visual change on their store. If this is intended, we'll need to be clear in the release notes. ([Before](https://screenshot.click/17-08-n56wi-1b6f3.png)) ([After](https://screenshot.click/17-08-12h6v-sylvm.png))
> * Opportunity to introduce the changes in 2 steps.
>   
>   * 1 is to keep relation ship with Products cards first regardless of the block type. Merchants could benefit from all the next theme releases with most of the great updates we are shipping to main.
>   * 2 would be to separate the associations relevant to the block type. Gives merchants the ability to chose or not to make the update yet and have already benefited from the pass release wins.
> * We may ok with this change because of the flexibility it might bring but merchants will have other key decisions to make in order to make the Collage section feel like it was stilled before. e.g. Settings applied to not only the Collage section and impacting other section where it may be unwanted.

👋 @melissaperreault, my understanding when I spoke to @duygukalaycioglu and @Oliviammarcello is that it is expected that the collage section will each get the styles from a different setting now 

Maybe it makes sense to update the release notes 🤔 I agree cc @katycobb

File: assets/component-card.css
Comment (andrewetchen) [reply assets/component-card.css:61]: Thanks Sofia. And in that case, we could add `*` to:

```css
.card * {
  text-decoration: none;
  text-align: var(--text-alignment);
}
```

It's a small improvement we can make to reduce a couple lines of CSS by not having to repeat the text alignment twice. I don't have strong feelings so I'll leave it up to you 😄

File: config/settings_data.json
Comment (andrewetchen) [reply config/settings_data.json:77]: Thanks for actioning this, Sofia! I'm going to "unresolve" this conversation simply for visibility 👍🏼

File: snippets/article-card.liquid
Comment (andrewetchen) [inline snippets/article-card.liquid:39]: > We'll need to consider associating a theme setting to apply to [Pages for the search results page](https://screenshot.click/16-41-g8q0p-l09jf.png). My recommendation would be to follow Blog posts for now. I know we need to improve those cards #1509 but that can be done later. We would at least give them a home for better grid results.

My interpretation of @melissaperreault's recommendation is to associate Pages within search results with the Blog cards settings until we have new theme settings specifically for Pages. This way, both Pages and Blogs within the search results will have better consistency.

Right now, Pages within the search results are using the Product cards settings:

**Review by andrewetchen — COMMENTED** [review]
Everything is looking good 👍🏼 

One thing that I also think we should address is what Melissa shared here: https://github.com/Shopify/dawn/pull/1774#issuecomment-1158459276

File: snippets/article-card.liquid
Comment (sofiamatulis) [reply snippets/article-card.liquid:39]: I think that makes sense 🤔 I will let @duygukalaycioglu confirm but also this isnt in scope of this PR and I can address it in a follow up if that is ok 👍

**Review by andrewetchen — DISMISSED** [review]
Thanks taking care of this, @sofiamatulis - everything looks good and works as expected 🎉 

Thanks for confirming with @duygukalaycioglu and @melissaperreault on the things we should tackle in a follow up 👍🏼 

*I'll re-approve after all translations are added

**Review by duygukalaycioglu — APPROVED** [review]

File: assets/collage.css
Comment (kmeleta) [inline assets/collage.css:113]: UX has probably given this more thought than me, but I kind of feel like collage cards should just be their own thing, either via global "collage card" settings or directly in the collage sections settings.

This current approach more or less makes it so either the merchant can't use collage sections OR can't reasonably make use of independent card type settings without making things ugly. Maybe with the occasional exception for the intentionally eclectic aesthetic. Perhaps that's a reasonable choice, but it seems ideal that we wouldn't force that choice on them.

I apologize if this has been addressed and I'm just missing some context.

File: assets/base.css
Comment (kmeleta) [inline assets/base.css:157]: For search results I see discussion about the page card types, but also wanted to make sure it's clear that this page could have a similar result to collage, where blog, product, and collection cards can all appear adjacent to each other in the same grid.


Unless all the card types are configured the same, this will _usually_ create an ugly result. And unlike collage, I don't really have any ideas on how to feasibly avoid it. Are we ok with this?

**Review by kmeleta — COMMENTED** [review]
Seems to be working great and the code is looking good. I mostly just wanted to raise a couple UX considerations, just to make sure we're all aligned on the trade offs. Sorry if any of it is redundant cc @melissaperreault @duygukalaycioglu

File: assets/collage.css
Comment (duygukalaycioglu) [reply assets/collage.css:118]: Thanks for the heads up, could you share one visual example where it breaks?

File: assets/collage.css
Comment (kmeleta) [reply assets/collage.css:113]: Indeed they can always choose to align the styles so collage cards don't visually clash, and that aligns with our stance of "they always have options to make things not ugly." But that's where I see it being an either-or sort of option for the merchant. You can't feasibly use a collage (with product and collection) AND use different card styles. Sounds like we're ok with this though, I just wanted to present potential options to allow both of this configurations at the same time.

**Review by kmeleta — COMMENTED** [review]
Empty collage product doesn't receive card settings


Empty collage media doesn't receive media or card settings


Non-empty collage media doesn't receive media or card settings

**Comment by sofiamatulis** [comment]
> Empty collage product doesn't receive card settings <img alt="" width="300" 

Good catch on the empty collage product! Fixed it. For the media, it isnt in scope for this PR since this is only affecting cards

File: assets/collage.css
Comment (kmeleta) [inline assets/collage.css:134]: > For the media, it isnt in scope for this PR since this is only affecting cards

Sorry just one last question. Is this documented in a fast follow issue yet? Because the media were originally using card styles, this introduces a regression as I see it.

And forgive me if I missed this somewhere, is the plan to use the global media settings for collage media cards or will they be inheriting from another card type?

Comment (sofiamatulis) [reply assets/collage.css:134]: @duygukalaycioglu Can we make sure we track this in a fast follow for a follow up ticket? 👀 Also wondering what this should be following 🤔 Collection card or product card

**Review by kmeleta — COMMENTED** [review]
As discussed, this looks good to go 👍  but I believe the change to collage media needs to be considered a regression and shouldn't be merged until that work is completed in another PR.

File: assets/collage.css
Comment (sofiamatulis) [reply assets/collage.css:134]: https://github.com/Shopify/dawn/issues/1822 issue created here

**Review by duygukalaycioglu — DISMISSED** [review]
The design and developers met on Thursday and aligned on:

- Cards on search page will be tackled as a follow up ticket [1823](https://github.com/Shopify/dawn/issues/1823). 
- Collage section will **globally** be tied to collection and product cards global theme settings. 
- Additional **section level setting** to be added to the collage section will be tackled as a follow up ticket [1826](https://github.com/Shopify/dawn/issues/1826). 
cc: @kmeleta @andrewetchen 

- [x] I'm approving the design task of breaking cards into blog, collection, and product cards global theme settings. This task doesn't include the new settings per card type, and doesn't include search page cards settings, and collage section level setting. 

- [x] I'm also approving collage section **global setting** above, which I reviewed few days ago. 

Thank you! @sofiamatulis This looks good to go.

**Review by katycobb — DISMISSED** [review]
Approving the global setting labels + order!

**Comment by duygukalaycioglu** [comment]
@sofiamatulis can we add one more global setting please, thanks!

- [ ] Media cards will be tied to product card settings globally. 

**Why?** This a decision made during the Thursday's alignment meeting. The media padding issue discussed here will be resolved [in this separate ticket](https://github.com/Shopify/dawn/issues/1826) when we add collage section level setting. If you have any questions please feel free to dm me @andrewetchen.

**Comment by sofiamatulis** [comment]
> in this separate ticket

@duygukalaycioglu to clarify, **only the media cards in collage** correct?

**Comment by duygukalaycioglu** [comment]
> @duygukalaycioglu to clarify, **only the media cards in collage** correct?

@sofiamatulis please see latest stakeholder approved decision below. 

Media = image and video

- [ ]  Cards with media (product cards, collection cards, blog cards will be tied to product card settings globally.

– Standalone media (blog pages, image with text) will continue to be tied to the global media settings under theme settings. No change to its current behavior. 
global media setting

File: sections/main-search.liquid
Comment (andrewetchen) [inline sections/main-search.liquid:233]: https://github.com/Shopify/dawn/pull/1774/conflicts

**Review by andrewetchen — COMMENTED** [review]
Everything is working great!

I'll approve once the following conflict is resolved 👍🏼

**Review by andrewetchen — APPROVED** [review]
Everything is working great! Thanks again for these changes, @sofiamatulis and @duygukalaycioglu

**Review by LucasLacerdaUX — APPROVED** [review]
Everything looks good to me :razz-thumbsup:

**Pull request merged**
